frontend/src/utilities/debounce.js

```diff
@@ -0,0 +1,25 @@
+// https://www.educative.io/edpresso/how-to-use-the-debounce-function-in-javascript
+
+function debounce(func, wait, immediate) {
+  var timeout;
+
+  return function executedFunction() {
+    var context = this;
+    var args = arguments;
+
```

Comment:
Great work, but I'd rather you bring in the lodash library and use the debounce function they provide instead of this snippet of code.  The lodash library is heavily tested.  Also, I recently worked on a change to have the keyword typed into the search to be stored into the redux store, so you might need to fix some conflicts.

Reply:
ok, then shall I go ahead with lodash.